### ISSUE TITLE
NumPy installation via setup.py fails

### ISSUE THREAD

**Issue by bnjones** [mathics/Mathics#418]
NumPy installation through setup.py fails in a clean virtualenv:

```
Processing numpy-1.11.1rc1.tar.gz
Writing /tmp/easy_install-sS5goJ/numpy-1.11.1rc1/setup.cfg
Running numpy-1.11.1rc1/setup.py -q bdist_egg --dist-dir /tmp/easy_install-sS5goJ/numpy-1.11.1rc1/egg-dist-tmp-3FqWrT
Running from numpy source directory.
/tmp/easy_install-sS5goJ/numpy-1.11.1rc1/setup.py:327: UserWarning: Unrecognized setuptools command, proceeding with generating Cython sources and expanding templates
  warnings.warn("Unrecognized setuptools command, proceeding with "
Warning: distutils distribution has been initialized, it may be too late to add a subpackage compat
[...]
```

Installing numpy with pip works fine. Looks like this could be the same issue as numpy/numpy#2434.

**Comment by sn6uv** [comment]
It works for me (with both python2 and python3).

Since numpy should be optional anyway I think should remove numpy from setup.py and only use it if it's already installed. What do you think @poke1024?

**Comment by bnjones** [comment]
Looking at the NumPy issue, it sounds like this could be something specific to Ubuntu. It looks like other projects are adding numpy to `setup_requires` as a workaround, I'll give this a try and open a PR if it works for me.

**Comment by poke1024** [comment]
Removing numpy from setup.py sounds ok to me, it's easy enough to install it manually (and components needing it in the future should give an easy to read error message in that regard), and breaking the standard installation in some environments is indeed bad. I've created https://github.com/mathics/Mathics/pull/423, in case the `setup_requires` solution @bnjones tries does not work.

**Comment by bnjones** [comment]
Unfortunately it looks like `setup_requires` won't really solve this problem - it doesn't install anything in the virtualenv, it just builds an egg in the current working directory for setup.py to use.

**Comment by bnjones** [comment]
Merged #423 to close this. Thanks!
